src/couch_replicator_clustering.erl

```diff
@@ -42,94 +59,79 @@ start_link() ->
 
 
 % owner/2 function computes ownership for a {DbName, DocId} tuple
-% Returns {ok no_owner} in case no DocId is null. That case
-% would happen in the old replicator_manager if replication was
-% posted from _replicate endpoint and not via a document in 
-% *_replicator db.
-%
-% {error, unstable} value is returned if cluster membership has
-% been changing recently. Recency is a configuration parameter.
+% Returns `no_owner` in case no DocId is null, `unstable` if cluster
+% is considered to be unstable i.e. it has changed recently, or returns
+% node() which is considered to be the owner.
 %
 -spec owner(Dbname :: binary(), DocId :: binary() | null) ->
```

Comment:
I'd like to simplify this, have a single owner function that takes a #rep{} or term(). In the case of unstable, we can return `nonode`, since we would do the same thing as if the job was owned by a foreign node (not start it if asked, stop it on next checkpoint).

The contract can be owner(#rep{} | term()) -> node() for all cases, we'll just sometimes return a node that is none of the running nodes.